### ISSUE TITLE
OBCs in restarts

### ISSUE THREAD

**Issue by kshedstrom** [ESMG/MOM6#76]
I've got a dumbbell case with a restart at time 0.1, so it's not painful to run both from time zero and from time 0.1. I can see that we register say rx_normal:
```
call register_restart_field(OBC%rx_normal, vd, .false., restart_CSp)
```
then in a call to radiation_open_bdry_conds we copy from OBC%rx_normal into a segment field. At that time, OBC%rx_normal is zero. What is the mechanism and timing of loading up these fields on restart?
@MJHarrison-GFDL @adcroft @Hallberg-NOAA

**Comment by kshedstrom** [comment]
Update on this. In MOM.F90, I had moved the call to open_boundary_register_restarts to after the call to MOM_initialize_state so that the DOME user code could set the tracer reservoir flags correctly first. That's what broke the restart stuff for the OBCs. Putting it back to before MOM_initialize_state fixed rx_normal.

As for the tracers, saving T and S if use_temperature is true is the easiest, rather than parsing MOM_input.

**Comment by kshedstrom** [comment]
So, the next restart issue is happening in remap_all_state_vars in ALE, where u and v get remapped. During the first timestep after a restart, the values outside the domain for h_old and h_new are 1.e-10 while later in the run they are 0.1. The remapping is using these bad outside values in an average to obtain h at the boundary. The OBC virus is spreading...

@MJHarrison-GFDL   @adcroft  @Hallberg-NOAA

**Comment by kshedstrom** [comment]
The ALE thing is fixed in #1015
